### PR TITLE
Preparation for adding the one wallet module

### PR DESCRIPTION
**Pull request by berryve**
Separate out common base and utility classes to common library
Create base classes for the various sample apps so code is not repeated

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 project.ext {
     minSdkVersion = 16
-    compileSdkVersion = 16
-    targetSdkVersion = 16
+    compileSdkVersion = 26
```
Comment (katzio) [inline build.gradle]: minSdkVersion & targetSdkVersion constants

File: build.gradle [hunk 2]
```diff
@@ -15,6 +15,7 @@ project.ext {
     testEspressoCoreVersion = '3.0.2'
 }
 buildscript {
+    ext.kotlin_version = '1.2.71'
     ext.kotlin_version = '1.1.51'
```
Comment (katzio) [inline build.gradle]: duplicated

File: baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/SampleBaseActivity.java [hunk 3]
```diff
@@ -1,37 +1,44 @@
...
 
         activityCreated = true;
 
         initAccountAndViews();
+        initCustomViews();
     }
 
-    private SampleWallet getSampleWallet() {
-        return ((ReceiverApplication) getApplicationContext()).getSampleWallet();
+    protected void initCustomViews() {
```
Comment (shayb) [inline baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/SampleBaseActivity.java]: not clear what is the difference between cusotom views and init account and view
maybe 2 methods 
init accounts
initviews

File: senderSampleApp/src/main/res/layout/activity_sample.xml [hunk 4]
```diff
@@ -47,5 +25,8 @@
         android:layout_marginTop="16dp"
         android:text="@string/show_alert"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/discoverBtn" />
+        app:layout_constraintTop_toBottomOf="@id/discoverBtn" />
+
+    <include layout="@layout/sample_base" />
```
Comment (shayb) [inline senderSampleApp/src/main/res/layout/activity_sample.xml:30]: very well!

**Review by shayb — APPROVED** [review]
looks great see my comment

File: baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/SampleBaseActivity.java (hunk 3 above)
Comment (berryve) [reply baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/SampleBaseActivity.java]: The purpose was to allow the extending class to add more views configuration/initialisation. Decided to remove the method and simply override onCreate from the extending class for that 😄

**Pull request merged**